File: 2-add-two-numbers/add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* temp1=l1;
        ListNode* temp2=l2;
        ListNode* dummy=new ListNode(0);
        ListNode* curr=dummy;
        int carry=0;
        while(temp1 != NULL || temp2!=NULL){
            int sum=carry;
            if(temp1){
                sum+=temp1->val;
            }
              if(temp2){
                sum+=temp2->val;
            }

            ListNode* add=new ListNode(sum%10);
            carry=sum/10;

            curr->next=add;
            curr=curr->next;

            if(temp1){
                temp1=temp1->next;
            }

            if(temp2){
                temp2=temp2->next;
            }
            if(carry){
                ListNode* add2=new ListNode(carry);
                curr->next=add2;
            }
        }    
        return dummy->next; 
    }
};

File: 2-add-two-numbers/add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        carry=0
        head=None
        current=None
        while l1!=None or l2!=None or carry!=0:
            if l1:
                val1=l1.val
                l1=l1.next
            else:
                val1=0
            if l2:
                val2=l2.val
                l2=l2.next
            else:
                val2=0              
            
            total=val1+val2+carry
            carry=total//10
            total=total%10
            # NewNode=ListNode()
            if head==None:
                head=ListNode(total)
                current=head
            else:
                current.next=ListNode(total)
                current=current.next
        return head
